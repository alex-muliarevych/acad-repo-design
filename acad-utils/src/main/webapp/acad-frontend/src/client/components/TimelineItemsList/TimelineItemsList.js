import React from 'react';
import PropTypes from 'prop-types';

// Custom components
import TimelineItem from 'components/TimelineItem';
import Popup from 'components/Popup';
import { consumePopup } from 'components/PopupContext';

// Helpers
import BEM from 'helpers/BEM';

// Styles
import './TimelineItemsList.less';

const bTimeItems = BEM.b('timeline-list');

@consumePopup
export default class TimelineItemsList extends React.Component {
  state = {
    availableTimeItems: []
  };

  componentDidMount() {
    this.setState({
      availableTimeItems: this.getAvailableTimeItems()
    });
  }

  /**
   * @description
   * On Popup close event callback listener
   * @memberof TimelineItemsList
   */
  onPopupClosedCallback = () => {
    console.log('Popup closed... Callback triggered!');
  }

  /**
   * @todo Change this mock data with redux store handlers
   *
   * @description
   * Creates array of hours for day to display in list for events creation
   *
   * @returns {Array} available hours
   * @memberof TimelineItemsList
   */
  getAvailableTimeItems() {
    const timeItems = [];

    for (let currentHour = 6; currentHour < 24; currentHour++) {
      timeItems.push({
        hour: currentHour
      });
    }

    return timeItems;
  }

  /**
   * @description
   * Renders list of timeline items based on available hours from state
   * @returns {React.Element<TimelineItem>[]} array of timeline items
   * @memberof TimelineItemsList
   */
  renderTimelineHours = () => {
    return (
      this.state.availableTimeItems.map(
        (item) => (
            <TimelineItem
              key={ item.hour }
              hour={ item.hour }>
              {/** @todo: remove this buttons when activities will be implemented
                 * Mock triggers to open popup for test
                 */}
              { [1, 2, 3].map((i) => (
                <button
                  key={ i }
                  onClick={ this.props.popups?.rightPanel.open }>
                  Show popup for activities #{ item.hour } - { i }
                </button>
              ))}
            </TimelineItem>
        ),
      )
    );
  }

  render() {
    return (
      <div className={ bTimeItems() }>
        <h2 className={ bTimeItems('heading') }>
          Today
        </h2>

        { this.renderTimelineHours() }

        <Popup instance={ this.props.popups?.rightPanel } >
          <h1>Right panel popup window</h1>
          <p>Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Aenean commodo ligula eget dolor.</p>
        </Popup>
      </div>
    );
  }
}
