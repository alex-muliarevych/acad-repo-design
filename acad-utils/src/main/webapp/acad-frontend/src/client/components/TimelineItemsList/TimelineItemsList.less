@import (reference) '~styles/variables.less';

.timeline-list {
  display: flex;
  flex-direction: column;
  flex: 2;
  padding: 14px;
  background-color: @light-grey;

  &__heading {
    margin-bottom: 30px;
  }
}
