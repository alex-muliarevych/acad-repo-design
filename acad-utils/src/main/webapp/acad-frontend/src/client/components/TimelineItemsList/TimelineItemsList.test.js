import React from 'react';
import Adapter from "enzyme-adapter-react-16";
import { shallow, configure } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import TimelineItemsList from './TimelineItemsList';

configure({ adapter: new Adapter() });

// Mock components
jest.mock('../TimelineItem/TimelineItem', () => 'mock-timeline-item');

describe('TimelineItemsList', () => {
  let component;
  
  beforeEach(() => {
    component = shallow(
      <TimelineItemsList />
    );
  });

  it(`should be truthy`, () => {
    expect(component).toBeTruthy();
  });

  it(`should render correctly`, () => {
    expect(shallowToJson(component)).toMatchSnapshot();
  });
});

