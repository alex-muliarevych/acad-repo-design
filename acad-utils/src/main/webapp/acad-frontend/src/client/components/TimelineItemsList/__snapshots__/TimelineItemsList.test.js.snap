// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimelineItemsList should render correctly 1`] = `null`;
